.github/workflows/pre_release_tests.yml

```diff
@@ -0,0 +1,53 @@
+name: pre release tests
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *'
+
+permissions:
+  contents: write
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+            python-version: 3.12
+      - name: Set up environment variables
+        working-directory: ./src
+        run: echo "Environment variables set up"
+        env:
+          CLIENT_ID: ${{ secrets.CLIENT_ID }}
+          CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+          BIO_API_KEY: ${{ secrets.BIO_API_KEY }}
+      - name: Install dependencies
+        run: |
+            python -m pip install --upgrade pip
+            pip install pytest requests~=2.32.3 requests-oauthlib~=2.0.0 linkml-runtime~=1.8.2 pyyaml~=6.0.2 tqdm~=4.66.5 pandas~=2.2.3 python-dotenv==1.0.1 typing-inspect~=0.9.0 nmdc_api_utilities
+            pip install --pre nmdc_schema
```

Comment:
Would this work if you just did
```
pip install --pre nmdc_schema
pip install -r requirements.txt
```

I don't want to have to remember to update the version number for the requirements here and in the requirements.txt.

Reply:
Maybe we'd need to specify nmdc-schema >= 12.4 or something like that in the requirements to make this work.

Reply:
Can we `pip install -r requirements.txt` then `pip uninstall nmdc-schema` then `pip install --pre nmdc_schema`?  Y